libretroshare/src/pqi/pqiservice.cc

```diff
@@ -204,40 +193,27 @@ bool p3ServiceServer::sendItem(RsRawItem *item)
 	}
 
 	mPublisher->sendItem(item);
+
 	return true;
 }
 
-
-
 int	p3ServiceServer::tick()
 {
-
 	mServiceControl->tick();
 
-	RsStackMutex stack(srvMtx); /********* LOCKED *********/
-
-#ifdef  SERVICE_DEBUG
-	pqioutput(PQL_DEBUG_ALL, pqiservicezone, 
-		"p3ServiceServer::tick()");
-#endif
-
-	std::map<uint32_t, pqiService *>::iterator it;
+	// make a copy of the service map
+	std::map<uint32_t,pqiService *> local_map;
+	{	
+		RS_STACK_MUTEX(srvMtx);
+		local_map=services;
```

Comment:
I remember discussing this, but now I have a doubt whether we actually need a mutex here. This map is not supposed to change accross time. I mean services are always the same.

Reply:
I dont know. I left the copy operation under mutex protection because the service map is non atomic.